src/server/controllers/FileCheckController.ts

```diff
@@ -0,0 +1,60 @@
+import { NextFunction, Request, Response } from 'express'
+import { inject, injectable } from 'inversify'
+
+import jsonMessage from '../util/json'
+import { FileCheckControllerInterface } from './interfaces/FileCheckControllerInterface'
+import { DependencyIds } from '../constants'
+import { FileTypeFilterServiceInterface } from '../services/interfaces/FileTypeFilterServiceInterface'
+import { VirusScanServiceInterface } from '../services/interfaces/VirusScanServiceInterface'
+
+@injectable()
+export class FileCheckController implements FileCheckControllerInterface {
+  private fileTypeFilterService: FileTypeFilterServiceInterface
+
+  private virusScanService: VirusScanServiceInterface
+
+  public constructor(
+    @inject(DependencyIds.fileTypeFilterService)
+    fileTypeFilterService: FileTypeFilterServiceInterface,
+    @inject(DependencyIds.virusScanService)
+    virusScanService: VirusScanServiceInterface,
+  ) {
+    this.fileTypeFilterService = fileTypeFilterService
+    this.virusScanService = virusScanService
+  }
+
+  public checkFile: (
+    req: Request,
+    res: Response,
+    next: NextFunction,
+  ) => Promise<void> = async (req, res, next) => {
+    const file = req.files?.file
+
+    if (Array.isArray(file)) {
+      res.badRequest(jsonMessage('Only single file uploads are supported.'))
```

Comment:
consider using `422 UNPROCESSABLE ENTITY`

Reply:
This was a copypaste from elsewhere in the codebase. I'll update that part too